cmake/templates/HPXMacros.cmake.in

```diff
@@ -11,30 +11,60 @@ include(GNUInstallDirs)
 include(HPX_Utils)
 
 function(hpx_check_compiler_compatibility)
-    if(HPX_IGNORE_COMPILER_COMPATIBILITY OR HPX_WITH_BUILD_BINARY_PACKAGE)
-    return()
-  endif()
 
-  if(NOT (${CMAKE_CXX_COMPILER_ID} STREQUAL ${HPX_CXX_COMPILER_ID}
-     AND ${CMAKE_CXX_COMPILER_VERSION} STREQUAL ${HPX_CXX_COMPILER_VERSION}))
-    set(MESSAGE "Compilers do not match. In order to compile HPX application it is"
-        " recommended to use the same compiler as you did for HPX. "
-        "HPX_CXX_COMPILER=${HPX_CXX_COMPILER}, "
-        "CMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}. "
-        "HPX_CXX_COMPILER_ID=${HPX_CXX_COMPILER_ID}, "
-        "CMAKE_CXX_COMPILER_ID=${CMAKE_CXX_COMPILER_ID}. "
-        "HPX_CXX_COMPILER_VERSION=${HPX_CXX_COMPILER_VERSION}, "
-        "CMAKE_CXX_COMPILER_VERSION=${CMAKE_CXX_COMPILER_VERSION}."
-        "To disable this message set HPX_IGNORE_COMPILER_COMPATIBILITY to On."
-        )
-    if (HPX_FIND_REQUIRED)
-      hpx_error("${MESSAGE}")
+    if(HPX_IGNORE_COMPILER_COMPATIBILITY)
+       return()
+    endif()
+       
+    if(HPX_WITH_BUILD_BINARY_PACKAGE)
+    
+        string(REGEX MATCH "[0-9]+" HPX_MAJOR_VERSION "${HPX_CXX_COMPILER_VERSION}")
+    	string(REGEX MATCH "[0-9]+" CXX_MAJOR_VERSION "${CMAKE_CXX_COMPILER_VERSION}")
```

Comment:
There's still something wonky here. The first `string(...` isn't aligned and the second has a tab. Sorry to be so annoying with this. Check that your editor isn't giving you tabs by default. Some of the if-statements further down also have funny alignments.

Reply:
Should be good now.